src/algorithms/digi/PhotoMultiplierHitDigi.cc

```diff
@@ -111,66 +115,69 @@ std::vector<edm4eic::MCRecoTrackerHitAssociation*> eicrecon::PhotoMultiplierHitD
         if(m_log->level() <= spdlog::level::trace) {
+          m_log->trace("{:-<70}","Accepted hit groups ");
           for(auto &[id,hitVec] : hit_groups)
             for(auto &hit : hitVec) {
-              m_log->trace("hit_group: pixel id={:#X} -> npe={} signal={} time={}", id, hit.npe, hit.signal, hit.time);
-              for(auto &mc_hit : hit.mc_hits)
-                m_log->trace("           - photon: EDep = {}", mc_hit->getEDep());
+              m_log->trace("hit_group: pixel id={:#018X} -> npe={} signal={} time={}", id, hit.npe, hit.signal, hit.time);
+              for(auto i : hit.sim_hit_indices)
+                m_log->trace(" - MC hit: EDep={}, id={}", sim_hits->at(i).getEDep(), sim_hits->at(i).id());
             }
         }
 
         // build output `MCRecoTrackerHitAssociation`
-        std::vector<edm4eic::MCRecoTrackerHitAssociation*> hit_assocs;
+        m_log->trace("{:-<70}","Digitized raw hits ");
+        PhotoMultiplierHitDigiResult result;
+        result.raw_hits   = std::make_unique<edm4eic::RawTrackerHitCollection>();
+        result.hit_assocs = std::make_unique<edm4eic::MCRecoTrackerHitAssociationCollection>();
         for (auto &it : hit_groups) {
             for (auto &data : it.second) {
 
                 // build `RawTrackerHit`
-                edm4eic::MutableRawTrackerHit raw_hit;
+                auto raw_hit = result.raw_hits->create();
                 raw_hit.setCellID(it.first);
                 raw_hit.setCharge(    static_cast<decltype(edm4eic::RawTrackerHitData::charge)>    (data.signal)                    );
                 raw_hit.setTimeStamp( static_cast<decltype(edm4eic::RawTrackerHitData::timeStamp)> (data.time/m_cfg.timeResolution) );
                 // raw_hit.setPosition(pos2vec(data.pos)) // TEST gap cuts; FIXME: requires member `edm4hep::Vector3d position`
                                                           // in data model datatype, think of a better way
-                m_log->trace("raw_hit: cellID={:#X} -> charge={} timeStamp={}",
+                m_log->trace("raw_hit: cellID={:#018X} -> charge={} timeStamp={}",
                     raw_hit.getCellID(),
                     raw_hit.getCharge(),
                     raw_hit.getTimeStamp()
                     );
 
                 // build `MCRecoTrackerHitAssociation`
-                edm4eic::MutableMCRecoTrackerHitAssociation hit_assoc;
+                auto hit_assoc = result.hit_assocs->create();
                 hit_assoc.setWeight(1.0); // not used
                 hit_assoc.setRawHit(raw_hit);
-                for(auto &mc_hit : data.mc_hits)
-                  hit_assoc.addToSimHits(*mc_hit);
-                hit_assocs.push_back(new edm4eic::MCRecoTrackerHitAssociation(hit_assoc)); // force immutable
+                for(auto i : data.sim_hit_indices)
+                  hit_assoc.addToSimHits(sim_hits->at(i));
             }
         }
-        return hit_assocs;
+        return std::move(result);
```

Comment:
https://www.youtube.com/watch?t=2284

Reply:
Good prank... my music player was having problems, then I clicked this link and thought I was having issues with Youtube too... anyway I'll drop the unnecessary `std::move`.